internal/controller/clusterlogging/clusterlogging_controller.go

```diff
@@ -0,0 +1,163 @@
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	// Fetch associated logging.ClusterLogForwarder if any
+	clf, err := loader.FetchClusterLogForwarder(r.Client, request.NamespacedName.Namespace, request.NamespacedName.Name)
+	if err != nil && !errors.IsNotFound(err) {
+		return ctrl.Result{}, err
+	}
+
+	// Check if fluentDForward is used & validates output secrets
+	outputSecrets, status, err := clusterlogforwarder.ValidateClusterLogForwarderForConversion(clf, r.Client)
+	if err != nil {
+		clf.Status = *status
+		if result, err := r.updateStatus(instance); err != nil {
+			return result, err
+		}
+		return ctrl.Result{}, err
+	}
+
+	// Convert to observability.ClusterLogForwarder
+	obsClf := api.ConvertLoggingToObservability(r.Client, instance, clf, outputSecrets)
+	// Fix indices for default elasticsearch to be `app-write`, `infra-write`, `audit-write`
+	obsClf.Spec, _ = obsMigrate.MigrateDefaultElasticsearch(obsClf.Spec)
+
+	if err = r.Client.Create(context.TODO(), obsClf); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// Annotate CL with migrated
+	instance.Annotations[constants.AnnotationCRConverted] = "true"
+	if err = r.Client.Update(context.TODO(), instance); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// Set and update status
+	instance.Status = loggingv1.ClusterLoggingStatus{}
+	instance.Status.Conditions.SetCondition(conditions.CondReadyWithMessage(loggingv1.ReasonMigrated, "logging.ClusterLogging migrated to observability.ClusterLogForwarder"))
+	if result, err := r.updateStatus(instance); err != nil {
+		return result, err
+	}
+
+	return ctrl.Result{}, err
+}
+
+func (r *ReconcileClusterLogging) updateStatus(instance *loggingv1.ClusterLogging) (ctrl.Result, error) {
+	if err := r.Client.Status().Update(context.TODO(), instance); err != nil {
+
+		if strings.Contains(err.Error(), constants.OptimisticLockErrorMsg) {
+			// do manual retry without error
+			// more information about this error here: https://github.com/kubernetes/kubernetes/issues/28149
+			return reconcile.Result{RequeueAfter: time.Second * 1}, nil
+		}
+
+		log.Error(err, "clusterlogging-controller error updating status")
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *ReconcileClusterLogging) SetupWithManager(mgr ctrl.Manager) error {
+	controllerBuilder := ctrl.NewControllerManagedBy(mgr).
+		For(&loggingv1.ClusterLogging{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		// Ignore create events as well as migrated resources
+		WithEventFilter(controller.IgnoreMigratedResources(constants.AnnotationCRConverted)).
+		Owns(&corev1.Service{}).
```

Comment:
I'm betting we don't need any of these Owns statements or Watches since we only care about the CL or CLF.  All the other related resources won't matter to us if they change since we are doing a conversion